bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java

```diff
@@ -454,15 +454,21 @@ public int available() {
     // Delegate all resumable operations to the scanner. It will request a non-resumable scanner
     // during operation.
     final ResponseQueueReader reader = new ResponseQueueReader(retryOptions.getReadPartialRowTimeoutMillis());
-    RetryingReadRowsOperation operation = createReadRowsRetryListener(request, reader);
-    operation.setResultObserver(new StreamObserver<ReadRowsResponse>(){
+    final StreamObserver<ReadRowsResponse> responseObserver = new StreamObserver<ReadRowsResponse>() {
       @Override
       public void onNext(ReadRowsResponse value) {
         reader.addRequestResultMarker();
       }
-      @Override public void onError(Throwable t) {}
-      @Override public void onCompleted() {}
-    });
+
+      @Override
+      public void onError(Throwable t) {
+      }
+
+      @Override
+      public void onCompleted() {
+      }
+    };
+    RetryingReadRowsOperation operation = createReadRowsRetryListener(request, reader, responseObserver);
```

Comment:
why does this need to insert a listener in the middle of the callback chain? Can the queue reader manage its own queue?

Reply:
I renamed this method to createReadRowsOperation.  Is that better?

Reply:
Yes, I think that's a better name